pkg/antctl/raw/bundle/command.go

```diff
@@ -0,0 +1,507 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/cheggaaa/pb/v3"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/time/rate"
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+
+	agentapiserver "github.com/vmware-tanzu/antrea/pkg/agent/apiserver"
+	"github.com/vmware-tanzu/antrea/pkg/agent/controller/noderoute"
+	"github.com/vmware-tanzu/antrea/pkg/antctl/runtime"
+	systemv1beta1 "github.com/vmware-tanzu/antrea/pkg/apis/system/v1beta1"
+	controllerapiserver "github.com/vmware-tanzu/antrea/pkg/apiserver"
+	antrea "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+)
+
+const (
+	barTmpl      pb.ProgressBarTemplate = `{{string . "prefix"}}{{bar . }} {{percent . }} {{rtime . "ETA %s"}}` // Example: 'Prefix[-->______] 20%'
+	requestRate                         = 100
+	requestBurst                        = 150
+)
+
+// Command is the bundle command implementation.
+var Command *cobra.Command
+
+var option = &struct {
+	dir           string
+	labelSelector string
+}{}
+
+var remoteControllerLongDescription = strings.TrimSpace(`
+	Generate support bundles of the cluster, includes: antrea agent bundles, antrea controller bundles and general information of the cluster.
+`)
+
+var remoteControllerExample = strings.Trim(`
+  Generate support bundle of all nodes and save them to current working dir
+  $ antctl bundle
```

Comment:
Is "bundle" common enough to represent "support bundle"? We are using it for package name, API name, command name but not sure if others can understand it.

Reply:
Migrated to term `support bundle`.